cli_tools/common/distro/distro.go

```diff
@@ -106,25 +132,38 @@ func FromComponents(distro string, major string, minor string) (r Release, e err
 //
 // https://cloud.google.com/sdk/gcloud/reference/compute/images/import#--os
 func FromGcloudOSArgument(osFlagValue string) (r Release, e error) {
-	os := strings.ToLower(osFlagValue)
-	if strings.HasSuffix(os, "-byol") {
-		os = strings.TrimSuffix(os, "-byol")
+	if osFlagValue == "windows-8-1-x64-byol" {
+		// windows-8-1-x64-byol is a legacy flag value, and it's the only value that
+		// includes an extra hyphen between its major and minor version. The non-legacy
+		// flag is windows-8-x64-byol.
+		osFlagValue = "windows-8-x64-byol"
+	}
```

Comment:
Constants. 

Maybe introduce a generic mapping if we have this issue in the future, e.g. a map from input OS flag value to "true" flag value.

Reply:
Switched to a map